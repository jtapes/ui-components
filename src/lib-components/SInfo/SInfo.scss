@import '../../theme/styles/variables';
@import '../../theme/styles/mixins';
@import '../../theme/styles/reset';
@import '../../theme/styles/fonts';
@import '../../theme/styles/global';

.s-info {
  position: relative;
  display: flex;
  align-items: center;
  padding: $size--8 $size--12;

  &__icon {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-right: $size--12;
    border-radius: 50%;
    width: $size--24;
    height: $size--24;
  }

  &--error {
    color: $color--danger;
    background-color: $color--danger-light;

    .s-info {
      &__icon {
        color: $color--white;
        background-color: $color--danger;
      }
    }
  }

  &--warning {
    color: $color--attention;
    background-color: $color--attention-light;

    .s-info {
      &__icon {
        color: $color--white;
        background-color: $color--attention;
      }
    }
  }

  &--success {
    color: $color--success;
    background-color: $color--success-light;

    .s-info {
      &__icon {
        color: $color--white;
        background-color: $color--success;
      }
    }
  }

  &--info {
    color: $color--primary-accent;
    background-color: $color--complimentary-bg-light;

    .s-info {
      &__icon {
        color: $color--white;
        background-color: $color--primary-accent;
      }
    }
  }
}
