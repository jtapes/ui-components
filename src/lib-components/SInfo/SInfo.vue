<template>
  <div class="s-info" :class="`s-info--${type}`">
    <div class="s-info__icon">
      <svg-icon :name="`${iconName}-16`" class="icon-16" />
    </div>
    <div class="text-sm">
      <!-- @slot Слот для сообщения  -->
      <slot>{{ message }}</slot>
    </div>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator'
import SvgIcon from '@/lib-components/SvgIcon/SvgIcon.vue'

const INFO_TYPE = ['error', 'warning', 'success', 'info']

/** Компонент для отображения сообщения */
@Component({
  components: {
    SvgIcon,
  }
})
export default class SInfo extends Vue {
  /** Тип сообщения */
  @Prop({
    type: String,
    validator(value: string): boolean {
      return INFO_TYPE.includes(value)
    },
  })
  type!: string

  /** Текст сообщения */
  @Prop(String)
  message?: string

  get iconName(): string | undefined {
    switch (this.type) {
      case 'error':
        return 'error'
      case 'warning':
        return 'info'
      case 'info':
        return 'info'
      case 'success':
        return 'checked'
    }
  }
}
</script>
