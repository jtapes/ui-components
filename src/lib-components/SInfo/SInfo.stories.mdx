import { action } from '@storybook/addon-actions'
import { config, withDesign } from 'storybook-addon-designs'
import { withTests } from '@storybook/addon-jest';
import {
  Meta,
  Story,
  ArgsTable,
  Description,
  Source,
  Props,
  Preview,
} from '@storybook/addon-docs'
import results from '/.jest-test-results.json';
import SInfo from '@/lib-components/SInfo/SInfo.vue'

<Meta
  title="DESIGN SYSTEM/Atom/Info"
  component={SInfo}
  decorators={[withDesign, withTests({results})]}
  argTypes={{
    message: { control: 'text' },
    type: {
      control: { type: 'select' },
      options: [
        'error',
        'warning',
        'success',
        'info',
      ],
      defaultValue: 'error'
    }
  }}
  parameters={{
    design: config({
      type: 'figma',
      url:
        'https://www.figma.com/file/42Xh2qI3qvq2CRkc2HJlwA/MVP-3-Market-Catalog-%E2%80%A2-Product-Page?node-id=1128%3A88',
    }),
    docs: {
      source: {
        type: 'code',
      },
    },
  }}
/>

# Информационный компонент

<Description of={SInfo} />

export const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { SInfo },
  template:
    `<s-info :message="message" :type="type" />`,
})

<Preview withToolbar withSource="none">
  <Story name="SInfo" args={{ message: 'Error', type: 'error' }}>
    {Template.bind({})}
  </Story>
</Preview>

## Пример

<Source
  language="html"
  code={`
    <s-info message="Error message" type="error" />
  `}
/>

<ArgsTable of={SInfo} />

