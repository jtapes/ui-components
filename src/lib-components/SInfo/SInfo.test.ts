import { mount } from '@vue/test-utils'
import SInfo from '@/lib-components/SInfo/SInfo.vue'

describe('SInfo', () => {
  test('Is Vue snapshot', () => {
    const wrapper = mount(SInfo, {
      propsData: {
        type: 'success',
        message: 'Test',
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
  test('Test error type', () => {
    const wrapper = mount(SInfo, {
      propsData: {
        type: 'error',
        message: 'Test',
      },
    })

    expect(wrapper.vm.$props.type).toBe('error')
  })
  test('Test warning type', () => {
    const wrapper = mount(SInfo, {
      propsData: {
        type: 'warning',
        message: 'Test',
      },
    })

    expect(wrapper.vm.$props.type).toBe('warning')
  })
})
