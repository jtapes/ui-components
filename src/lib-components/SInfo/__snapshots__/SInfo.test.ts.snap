// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SInfo Is Vue snapshot 1`] = `
<div
  class="s-info s-info--success"
>
  <div
    class="s-info__icon"
  >
    <svg
      class="svg-icon icon-16 svg-icon--checked-16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        href="#checked-16"
        xlink:href="#checked-16"
      />
    </svg>
  </div>
   
  <div
    class="text-sm"
  >
    Test
  </div>
</div>
`;
